pkg/platform/yaml/yaml.go

```diff
@@ -129,57 +130,81 @@ func (ys *Yaml) Write(path, fileName string, docRead platform.KindSpecifier) err
 	return nil
 }
 
-func containsMatchingUrl(urlMethods map[string][]string, urlStr string, requestMethod models.Method) bool {
-	parsedURL, err := url.Parse(urlStr)
+func containsMatchingUrl(urlMethods []string, urlStr string, requestUrl string, requestMethod models.Method) (error, bool) {
+	urlMatched := false
+	parsedURL, err := url.Parse(requestUrl)
 	if err != nil {
-		return false
+		return err, false
 	}
 
 	// Check for URL path and method
-	path := parsedURL.Path
-	if methods, exists := urlMethods[path]; exists {
-		// Loop through the methods for this path
-		for _, method := range methods {
-			// If the request method matches one of the allowed methods, return true
+	regex, err := regexp.Compile(urlStr)
+	if err != nil {
+		return err, false
+	}
+
+	urlMatch := regex.MatchString(parsedURL.Path)
+
+	if urlMatch && len(urlStr) != 0 {
+		urlMatched = true
+	}
+
+	if len(urlMethods) != 0 {
```

Comment:
Looks like this logic has flaw lets say user has given path - x and port -y . even if there is a request with path - z and port - y it will be matched and considered as passthrough.

Reply:
seems like this selected code block deals with some other logic.